src/core/common/message.cpp

```diff
@@ -91,6 +91,24 @@ Message *MessagePool::New(uint8_t aType, uint16_t aReserved, uint8_t aPriority)
     return message;
 }
 
+Message *MessagePool::New(uint8_t aType, uint16_t aReserved, otMessageSettings *aSettings)
+{
+    Message *message = NULL;
+    bool     linkSecurityEnabled;
+    uint8_t  priority;
+
+    linkSecurityEnabled = (aSettings == NULL) ? true : aSettings->mLinkSecurityEnabled;
```

Comment:
Have a single if-statement for `aSettings == NULL` check?

Reply:
Will update it.